### DISCUSSION TITLE
There is some delay in popup mode BEX

### DISCUSSION THREAD

**Discussion by pratik227** [quasarframework/quasar#15447] [CLI - BEX mode]
There is some delay while BEX popup open it first display blank screen and after some time it displays the actual content.

<img width="210" alt="Screenshot 2023-02-15 at 5 02 22 PM" src="https://user-images.githubusercontent.com/34883558/219016458-24b8593b-4e6a-4713-93c6-edc10af66b77.png">

it looks like this and after 1 or 2 seconds it displays actual content is there any way to fix this ?

I'm using Quasar 2 with Vite and manifest_version 3

**Comment by NaderIkladious** [comment]
I have the same issue, I couldn't understand why

I event created an [issue](https://github.com/quasarframework/quasar/issues/16168) for it but no responses so far

**Comment by hazaveh** [comment]
I am experiencing the same. Is there any workaround to make it faster?

**Comment by richterdennis** [comment]
Checkout this file:
https://github.com/quasarframework/quasar/blob/dbebdc52574e59d6853c1a004cbca7ea91a9b7e7/app-vite/templates/entry/client-entry.js#L234

There is the following code:
```
if (chrome.runtime.id) {
  // Chrome ~73 introduced a change which requires the background connection to be
  // active before the client this makes sure the connection has had time before progressing.
  // Could also implement a ping pattern and connect when a valid response is received
  // but this way seems less overhead.
  setTimeout(() => {
    connect()
  }, 300)
}
```

If u search ur node_modules folder for the phrase `chrome.runtime.id` u find this. Change the 300 to 0 and ur popup opens instant. This is a dirty fix for sure :D

**Reply by AleLuppi** [reply]
Rather than handling `node_modules`, I'd prefer to modify `dist` files only. The timeout can be found under `dist/bex/www/assets/index.*.js`, however it should be re-set at each bex update...

BTW, is this still required on newer Chrome version? If yes (which I expect), then I think the "ping" way should finally come to life.
